src/event.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { YGMeasureMode } from './enums';
+import { YGConfig } from './ygconfig';
+import { YGNode } from './ygnode';
+
+export enum LayoutType {
+    kLayout = 0,
+    kMeasure = 1,
+    kCachedLayout = 2,
+    kCachedMeasure = 3,
+}
+
+export enum LayoutPassReason {
+    kInitial = 0,
+    kAbsLayout = 1,
+    kStretch = 2,
+    kMultilineStretch = 3,
+    kFlexLayout = 4,
+    kMeasureChild = 5,
+    kAbsMeasureChild = 6,
+    kFlexMeasure = 7,
+}
+
+type LayoutPassCounts = [number, number, number, number, number, number, number, number];
+
+export class LayoutData {
+    layouts: number;
+    measures: number;
+    maxMeasureCache: number;
+    cachedLayouts: number;
+    cachedMeasures: number;
+    measureCallbacks: number;
+    measureCallbackReasonsCount: LayoutPassCounts;
+
+    constructor(
+        layouts = 0,
+        measures = 0,
+        maxMeasureCache = 0,
+        cachedLayouts = 0,
+        cachedMeasures = 0,
+        measureCallbacks = 0,
+        measureCallbackReasonsCount: LayoutPassCounts = [0, 0, 0, 0, 0, 0, 0, 0],
+    ) {
+        this.layouts = layouts;
+        this.measures = measures;
+        this.maxMeasureCache = maxMeasureCache;
+        this.cachedLayouts = cachedLayouts;
+        this.cachedMeasures = cachedMeasures;
+        this.measureCallbacks = measureCallbacks;
+        this.measureCallbackReasonsCount = measureCallbackReasonsCount;
+    }
```

Comment:
just for code improving purposes you could make it much smaller by using a `public` keyword in constructor
```suggestion
    constructor(
        public layouts = 0,
        public measures = 0,
        public maxMeasureCache = 0,
        public cachedLayouts = 0,
        public cachedMeasures = 0,
        public measureCallbacks = 0,
        public measureCallbackReasonsCount: LayoutPassCounts = [0, 0, 0, 0, 0, 0, 0, 0],
    ) {}
```